romancal/datamodels/models/image/dqflags.py

```diff
@@ -0,0 +1,150 @@
+         'UNRELIABLE_ERROR': 2**8,   # Uncertainty exceeds quoted error
+         'NON_SCIENCE':      2**9,   # Pixel not on science portion of detector
+         'DEAD':             2**10,  # Dead pixel
+         'HOT':              2**11,  # Hot pixel
+         'WARM':             2**12,  # Warm pixel
+         'LOW_QE':           2**13,  # Low quantum efficiency
+         'TELEGRAPH':        2**15,  # Telegraph pixel
+         'NONLINEAR':        2**16,  # Pixel highly nonlinear
+         'BAD_REF_PIXEL':    2**17,  # Reference pixel cannot be used
+         'NO_FLAT_FIELD':    2**18,  # Flat field cannot be measured
+         'NO_GAIN_VALUE':    2**19,  # Gain cannot be measured
+         'NO_LIN_CORR':      2**20,  # Linearity correction not available
+         'NO_SAT_CHECK':     2**21,  # Saturation check not available
+         'UNRELIABLE_BIAS':  2**22,  # Bias variance large
+         'UNRELIABLE_DARK':  2**23,  # Dark variance large
+         'UNRELIABLE_SLOPE': 2**24,  # Slope variance large (i.e., noisy pixel)
+         'UNRELIABLE_FLAT':  2**25,  # Flat variance large
+         'RESERVED_5':       2**26,  #
+         'RESERVED_6':       2**27,  #
+         'UNRELIABLE_RESET': 2**28,  # Sensitive to reset anomaly
+         'RESERVED_7':       2**29,  #
+         'OTHER_BAD_PIXEL':  2**30,  # A catch-all flag
+         'REFERENCE_PIXEL':  2**31,  # Pixel is a reference pixel
+}
+
+
+# Group-specific flags. Once groups are combined, these flags
+# are equivalent to the pixel-specific flags.
+group = {'GOOD':       pixel['GOOD'],
+         'DO_NOT_USE': pixel['DO_NOT_USE'],
+         'SATURATED':  pixel['SATURATED'],
+         'JUMP_DET':   pixel['JUMP_DET'],
+         'DROPOUT':    pixel['DROPOUT'],
+}
+
+
+def interpret_bit_flags(bit_flags, flip_bits=None):
+    """Converts input bit flags to a single integer value (bit mask) or `None`.
+
+    Wraps `astropy.nddate.bitmask.interpret_bit_flags`, allowing the Roman
+    bit mnemonics to be used in place of integers.
+
+    Parameters
+    ----------
+    bit_flags : int, str, list, None
+        See `astropy.nddate.bitmask.interpret_bit_flags`.
+        Also allows strings using Roman mnemonics
+
+    flip_bits : bool, None
+        See `astropy.nddate.bitmask.interpret_bit_flags`.
+
+    Returns
+    -------
+    bitmask : int or None
+        Returns an integer bit mask formed from the input bit value or `None`
+        if input ``bit_flags`` parameter is `None` or an empty string.
+        If input string value was prepended with '~' (or ``flip_bits`` was set
+        to `True`), then returned value will have its bits flipped
+        (inverse mask).
+
+    Examples
+    --------
+    Using Roman mnemonics:
+    TBD
+    """
+    bit_flags_dm = bit_flags
+    if isinstance(bit_flags, str):
+        dm_flags = {
+            key: str(val)
+            for key, val in pixel.items()
+        }
+        bit_flags_dm = multiple_replace(bit_flags, dm_flags)
```

Comment:
Do we need to worry about flag names that are substrings of other flag names?  If so, we ought to add `r'\b'` to the beginning and ending of each flag.

Reply:
Never mind!  I didn't notice that the code in `multiple_replace` sorts the keys with the longest first.